crates/publish/src/bindle_writer.rs

```diff
@@ -61,23 +60,23 @@ impl BindleWriter {
         let parcel_writes = parcels
             .iter()
             .map(|parcel| self.write_one_parcel(parcels_dir, parcel));
+
         futures::future::join_all(parcel_writes)
             .await
             .into_iter()
-            .collect::<anyhow::Result<Vec<_>>>()?;
+            .collect::<PublishResult<Vec<_>>>()?;
+
         Ok(())
     }
 
-    async fn write_one_parcel(&self, parcels_dir: &Path, parcel: &Parcel) -> Result<()> {
+    async fn write_one_parcel(&self, parcels_dir: &Path, parcel: &Parcel) -> PublishResult<()> {
         let source_file = match self.parcel_sources.source(&parcel.label.sha256) {
             Some(path) => path.clone(),
             None => self.source_dir.join(&parcel.label.name),
         };
         let hash = &parcel.label.sha256;
         let dest_file = parcels_dir.join(format!("{}.dat", hash));
-        tokio::fs::copy(&source_file, &dest_file)
-            .await
-            .with_context(|| copy_parcel_failed_msg(&source_file, &dest_file))?;
+        tokio::fs::copy(&source_file, &dest_file).await?;
```

Comment:
I worry about the loss of context here.  It looks like if a copy fails, we will no longer get a "parcel copy failed" message, but instead an `Io(io::Error)` which will not tell us what was happening when we failed or what file we failed on.  Am I missing something?

Reply:
Is there a use-case when copying only one of the parcels will fail?

I was trying to find a balance between wrapping on every level and not giving enough info for debug. Because to my taste, wrapping on every level might be a bit excessive, for ex.:

```
/tmp/spin bindle push --bindle-server http://localhost:8000/v1
Error: Failed to push bindle to server

Caused by:
    0: Failed to push bindle from '/tmp/.tmpaDtdJm' to server at 'http://localhost:8000/v1'
    1: Error creating request
    2: error sending request for url (http://localhost:8000/v1/_i): error trying to connect: tcp connect error: Connection refused (os error 111)
    3: error trying to connect: tcp connect error: Connection refused (os error 111)
    4: tcp connect error: Connection refused (os error 111)
    5: Connection refused (os error 111)
```

Reply:
No, but there is value in telling the user _which_ parcel failed.  For example, if a file is missing or permission denied or something, I need to know _which_ file so I can go in and fix it; I don't want to scrobble through every file in the project.

I agree we sometimes go overboard on error wrapping, and I'm happy to see less of that in the main output (e.g. we could consign the full context to logs, though we'd want a nicer experience for displaying them in this case, a la Fisher's `-vvvvvv`).  But I don't want to lose information that is critical to diagnosing and remedying a failure.  We got burned by "file does not exist (os error 2)" and NOT KNOWING WHICH FILE far too often in the early days of Spin...!

Hope that makes sense!

Reply:
done